napari/movie/naparimovie.py

```diff
@@ -0,0 +1,364 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+import imageio
+import copy
+
+from . import util
+
+
+class Movie:
+    def __init__(self, myviewer=None, inter_steps=15):
+
+        """Standard __init__ method.
+
+        Parameters
+        ----------
+        myviewer : napari viewer
+            napari viewer
+        inter_steps: int
+            number of steps to interpolate between key frames
+
+        Attributes
+        ----------
+        key_frames : list
+            list of dictionary defining napari viewer states. Dictionaries have keys:
+                'frame': int, frame
+                'ndisplay': int, view type (2,3)
+                'camera': camera state
+                'vis': list of booleans, visibility of layers
+                'sliders': list, slider positions
+
+        interpolated_states: dict
+            dictionary defining interpolated states. Each element is a list of length N
+            frames. Keys are:
+                'ndisplay': int, view type (2,3)
+                'camera': camera state
+                'vis': list of booleans, visibility of layers
+                'sliders': list, slider positions
+
+        states_dict : list
+            list of dictionaries defining napari viewer states for each frame. Same keys as key_frames
+
+        current_frame : int
+            currently shown key frame
+        implot : matplotlib Ax object
+            reference to matplotlib image used for movie returned by imshow
+        anim : matplotlib FuncAnimation object
+            reference to animation object
+        """
+
+        if myviewer is None:
+            raise TypeError(
+                "You need to pass a napari viewer for the myviewer argument"
+            )
+        else:
+            self.myviewer = myviewer
+
+        self.key_frames = []
+        self.inter_steps = inter_steps
+
+        self.current_frame = -1
+        self.current_interpolframe = 0
+
+        # establish key bindings
+        self.add_callback()
+
+    def finish_movie(self):
+        self.release_callbacks()
+
+    def add_callback(self):
+        """Bind keys"""
+
+        self.myviewer.bind_key("f", self.capture_keyframe_callback)
```

Comment:
I think I'd like to make these keybindings more complex than just presses, like "command+shift+f" so they don't conflict with inbuilt ones / ones likely to be in regular use inside plugins. I'm open to suggestions though

Reply:
One needs a lot of these key-presses when creating the key-frames and going back and forth within them, so this is why I'd prefer to keep them simple. Note that the bindings are released when movie recording is over so it should not conflict too much.

Reply:
fair enough, further down the line we'll maybe want to add buttons to the GUI to create one of these movie objects and then visually indicate in the GUI that keybindings have changed, and then add a button to finish movie making, we'll also want to add them to our keybindings info dialog (future PR). I'm happy to leave as is for now. We might also do a pretty major relook at all our keybindings at some point and try and establish standard patterns across them and so we can revisit at that point. What do @jni and @royerloic think?

Reply:
I like single stroke keybindings in general (again, mostly people will not be typing text into napari), and even more so in specific, short-lived modes.